libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -650,20 +657,19 @@ instance HasSpec fn a => Semigroup (Specification fn a) where
   MemberSpec as <> MemberSpec as' =
     explainSpec
       (NE.fromList ["Intersecting: ", "  MemberSpec " ++ show as, "  MemberSpec " ++ show as'])
-      $ MemberSpec
-      $ nub
-      $ intersect as as'
+      ( memberSpecList
+          (nub $ intersect (NE.toList as) (NE.toList as'))
+          (pure "Empty intersection combining two MemberSpecs")
+      )
```

Comment:
With the explanation above I feel like the message to `memberSpecList` is a bit redundant and will mostly create noise?

Reply:
How about  (pure "Empty intersection")